lib/aws/rails/sqs_active_job/executor.rb

```diff
@@ -36,8 +42,31 @@ def execute(message)
               job_msg = job ? "#{job.id}[#{job.class_name}]" : 'unknown job'
               @logger.info "Error processing job #{job_msg}: #{e}"
               @logger.debug e.backtrace.join("\n")
+            ensure
+              refresh_monitor = false
             end
           end
+
+          @monitor.post(message) do |message|
+            while refresh_monitor
+              begin
+                # Extend the visibility timeout to the provided refresh timeout
+                message.change_visibility({ visibility_timeout: @visibility_refresh })
+                # Wait half the refresh timeout, and repeat
+                sleep(@visibility_refresh / 2.0)
```

Comment:
why do we want to sleep? if I set this to 6 hours, a worker would be stalled for a while.

On the other hand, If we only are in this block when a message fails, then just changing the visibility will allow it to get picked up by a worker again